proposals/dark_mode_extension_icons.md

```diff
@@ -0,0 +1,205 @@
+# Dark Mode Extension Icons
+
+**Summary**
+
+Feature to enable developers to enhance extension icon visibility in dark mode.
+
+**Document Metadata**
+
+**Author:** &lt;solomonkinard>
+
+**Sponsoring Browser:** Chromium.
+
+**Contributors:** &lt;oliverdunk>
+
+**Created:** 2024-04-05
+
+**Related Issues:**
+* https://crbug.com/893175
+* https://github.com/w3c/webextensions/issues/229
+
+## Motivation
+
+### Objective
+
+Extension developers will be able to supply and define a set of icons to be used
+in the event that browser system settings are set to dark mode. The set of dark
+mode icons can and will be made automatically visible once the system setting is
+in dark mode, thereby enhancing the overall browser experience.
+
+#### Use Cases
+
+1. Improved icon visibility in the extension toolbar.
+1. Improved icon visibility on management the management and shortcuts pages.
+1. Dark mode icon declarations made possible through the extension manifest.
+1. setIcon() will allow setting of dark and/or light mode specific icons.
+1. Improved icon visibility on context menu (Chrome and Safari reply on default icons, only Firefox can specify icons).
+1. Improved icon visibility on side panel (Chrome replies on default icons, Firefox can specify icons).
+
+### Known Consumers
+
+The associated and linked bug has 48 stars and 31 comments.
+
+## Specification
+
+### Schema
+
+Some browsers don't currently have plans to support svg. Therefore, a drop-in
+replacement of an already supported image type will be allowed.
+
+numberToPathDictionary
+```
+{
+  "<number>": "<path>"
+}
+```
+
+manifest.json
+```
+{
+  "icon_variants": {
+    "<dark|light>": <numberToPathDictionary>
```

Comment:
I think to address @xeenon's feedback we should also accept a single icon here?

Reply:
Done.